geopandas/io/tests/test_file.py

```diff
@@ -76,7 +84,7 @@ def test_to_file(tmpdir, df_nybb, df_null, driver, ext):
     assert np.alltrue(df["BoroName"].values == df_nybb["BoroName"])
 
     # Write layer with null geometry out to file
-    tempfilename = os.path.join(str(tmpdir), "null_geom." + ext)
+    tempfilename = os.path.join(str(tmpdir), "null_geom" + ext)
     df_null.to_file(tempfilename, driver=driver)
```

Comment:
Do we check somewhere that we actually write the correct format? Because eg in this test we are reading it back, but that will also pass if we were still always writing shapefiles (if the detection would not have worked)

Reply:
Sounds like a good idea to me. I added `assert_correct_driver` to check.